hooks/tk-multi-publish2/ingest/ingest_files.py

```diff
@@ -0,0 +1,198 @@
+        # so that we can link the version created to plate entity as well.
+        item.properties["ingest_plate_data"] = plate_entity
+
+        # link the publish file to our plate entity.
+        self._link_published_files_to_plate_entity(plate_entity, item)
+        self.logger.info("Plate entity registered and Publish file is linked!")
+
+    def finalize(self, task_settings, item):
+        """
+        Execute the finalization pass. This pass executes once
+        all the publish tasks have completed, and can for example
+        be used to version up files.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the task_settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+
+        super(IngestFilesPlugin, self).finalize(task_settings, item)
+
+        # get the data for the plate that was just created in SG
+        plate_data = item.properties["ingest_plate_data"]
+
+        path = item.properties["path"]
+
+        self.logger.info(
+            "Plate created for file: %s" % (path,),
+            extra={
+                "action_show_in_shotgun": {
+                    "label": "Show Plate",
+                    "tooltip": "Open the Publish in Shotgun.",
+                    "entity": plate_data
+                }
+            }
+        )
+
+    @staticmethod
+    def _validate_plate_entity(item):
+        sg = get_sg_connection()
+        sg_filters = [
+            ['project', 'is', item.context.project],
+            ['code', 'is', item.properties["publish_name"]]
+        ]
+        if item.context.entity:
+            if item.context.entity["type"] == "Shot":
+                sg_filters.append(['shots', 'is', item.context.entity])
+            elif item.context.entity["type"] == "Sequence":
+                sg_filters.append(['sg_sequence', 'is', item.context.entity])
+
+        result = sg.find_one(
+            entity_type='Element',
+            filters=sg_filters,
+            fields=['shots', 'code', 'id']
+        )
+        return result
+
+    def _get_frame_range(self, item):
+        publisher = self.parent
+
+        # Determine if this is a sequence of paths
+        if item.properties["is_sequence"]:
+            first_frame = publisher.util.get_frame_number(item.properties["sequence_paths"][0])
+            last_frame = publisher.util.get_frame_number(item.properties["sequence_paths"][-1])
+        else:
+            first_frame = last_frame = 0
+
+        return first_frame, last_frame
+
+    def _create_plate_entity(self, item):
+        sg = get_sg_connection()
+        plate_entity = self._validate_plate_entity(item)
+        frange = self._get_frame_range(item)
+
+        data = dict(
+            code=item.properties["publish_name"],
+            sg_client_name=item.name,
+        )
+
+        data["cut_in"] = frange[0]
```

Comment:
You sure this should be cut_in/cut_out and not head_in/head_out?

Reply:
Yeah on the Element entity, those are the fields that the loader uses when creating the DCC nodes.

Reply:
Let's discuss this at our meeting today.

Reply:
Switching this to update head_in and head_out fields now, we will need to update loader to be giving priority to these keys from now on, will open a ticket for that too.